File: backend/src/api/games.py
"""
Game endpoints - save scores and analytics
"""

from fastapi import APIRouter, HTTPException, Depends
from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
from src.models.schemas import SaveScoreRequest, SaveScoreResponse
from src.services.game_service import GameService
from src.utils.database import get_db
from src.api.auth import get_current_user
from supabase import Client

router = APIRouter()
security = HTTPBearer()

@router.post("/save-score", response_model=SaveScoreResponse)
async def save_score(
    request: SaveScoreRequest,
    current_user: dict = Depends(get_current_user),
    db: Client = Depends(get_db)
):
    """Save game score and analytics"""
    game_service = GameService(db)
    result = await game_service.save_game_session(
        user_id=current_user["id"],
        game_id=request.gameId,
        analytics=request.analytics
    )
    
    if not result["success"]:
        raise HTTPException(status_code=500, detail=result.get("error", "Failed to save score"))
    
    return SaveScoreResponse(
        success=True,
        sessionId=result["sessionId"]
    )

File: backend/src/api/games.ts
/**
 * Game API Routes
 * 
 * This file contains the API route handlers for game-related endpoints.
 * In Next.js, these routes are implemented in frontend/app/api/games/
 * but the service logic is maintained here for separation of concerns.
 */

import { GameScoreService } from '../services/gameScoreService'
import { GameAnalytics } from '../types/game'

/**
 * Example API route handler for saving game scores
 * 
 * This is a reference implementation. The actual Next.js route
 * is in frontend/app/api/games/save-score/route.ts
 */
export async function saveGameScoreHandler(
  supabase: any,
  userId: string,
  gameId: string,
  analytics: GameAnalytics
) {
  const gameScoreService = new GameScoreService(supabase)
  
  try {
    const session = await gameScoreService.saveGameSession(userId, gameId, analytics)
    return { success: true, sessionId: session.id }
  } catch (error: any) {
    throw new Error(`Failed to save game score: ${error.message}`)
  }
}

